_posts/2021-04-05-fmf-unified-testing.md

```diff
@@ -0,0 +1,156 @@
+---
+title: Unified upstream/downstream testing with TMT and Packit
+author: pitti
+date: 2021-04-05
+category: howto
+tags: qa testing gating distribution fmf tmt fedora packit
+comments: true
+---
+
+Two and a half years ago, we started to [gate the Fedora cockpit package on our browser integration tests](https://src.fedoraproject.org/rpms/cockpit/c/f639e531f9357e1a42ce7a050726d520c1586535), and since then have run an ever-increasing number of tests in it. When doing it properly, automated package update gating can increase the quality of a Linux distribution tremendously -- I've worked a lot on that in [Ubuntu](https://wiki.ubuntu.com/ProposedMigration) and [Debian](https://ci.debian.net/status/) [many years ago](https://piware.de/2011/11/12-04-testing-ftw/), and I am really happy to see it happening in Fedora as well, even though adoption is slow (because it does not do reverse dependency gating yet).
```

Comment:
@dperpeet Do we know if someone in your team thinking of pulling off reverse dependency gating into Fedora? Alternatively is this something that Packit needs to cover for, as an added layer on top (request Packit to run tests of all dependent packages) because the the CI system is unable to?

Somehow I thought that adoption of Zuul by Fedora CI as a scheduler was meant to solve this.

Reply:
@stefwalter Yes, it's definitely on the roadmap. Fedora CI has plans for this, potentially the Testing Farm also. I'll see about a a sync between Packit and the other teams to see where there might be potential to adjust who does what. I haven't heard about Zuul doing this.
There are also discussions going on regarding how to best start and also resolve reverse dependency gating conversations. The ones that are running right now show that the involved teams need some more practice and likely guidance on how to deal with the failures that inevitably crop up.

Reply:
Running reverse dependency cating for upstream PRs in Packit-as-a-Service sounds a bit excessive (this scales up really quickly!). But for Fedora I believe it is the driving motivator for adding tests at all -- so far, the only result of adding more tests is to make your own life harder. But what you *really* want (the "carrot") is to make life of *other* people harder who try to upload stuff that breaks your product :grin: